src/libraries/System.Private.CoreLib/src/System/Threading/ThreadPool.Windows.cs

```diff
@@ -0,0 +1,228 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+
+using System.Diagnostics;
+using System.Runtime.CompilerServices;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+
+namespace System.Threading
+{
+    public static partial class ThreadPool
+    {
+        internal static bool UseWindowsThreadPool { get; } =
+            AppContextConfigHelper.GetBooleanConfig("System.Threading.ThreadPool.UseWindowsThreadPool", "DOTNET_ThreadPool_UseWindowsThreadPool");
+
+#if NATIVEAOT
+        private const bool IsWorkerTrackingEnabledInConfig = false;
+#else
+        private static readonly bool IsWorkerTrackingEnabledInConfig =
+            UseWindowsThreadPool ? false : AppContextConfigHelper.GetBooleanConfig("System.Threading.ThreadPool.EnableWorkerTracking", "DOTNET_ThreadPool_EnableWorkerTracking", false);
+#endif
+
+        // Indicates whether the thread pool should yield the thread from the dispatch loop to the runtime periodically so that
+        // the runtime may use the thread for processing other work.
+        //
+        // Windows thread pool threads need to yield back to the thread pool periodically, otherwise those threads may be
+        // considered to be doing long-running work and change thread pool heuristics, such as slowing or halting thread
+        // injection.
+        internal static bool YieldFromDispatchLoop => UseWindowsThreadPool;
+
+        [CLSCompliant(false)]
+        [SupportedOSPlatform("windows")]
+        public static unsafe bool UnsafeQueueNativeOverlapped(NativeOverlapped* overlapped) =>
+            ThreadPool.UseWindowsThreadPool ?
+            WindowsThreadPool.UnsafeQueueNativeOverlapped(overlapped) :
+            UnsafeQueueNativeOverlappedPortableCore(overlapped);
+
+        [Obsolete("ThreadPool.BindHandle(IntPtr) has been deprecated. Use ThreadPool.BindHandle(SafeHandle) instead.")]
+        [SupportedOSPlatform("windows")]
+        public static bool BindHandle(IntPtr osHandle) =>
+            ThreadPool.UseWindowsThreadPool ?
+            WindowsThreadPool.BindHandle(osHandle) :
+            BindHandlePortableCore(osHandle);
+
+        [SupportedOSPlatform("windows")]
+        public static bool BindHandle(SafeHandle osHandle) =>
+            ThreadPool.UseWindowsThreadPool ?
+            WindowsThreadPool.BindHandle(osHandle) :
+            BindHandlePortableCore(osHandle);
+
+#if !CORECLR
+        private static bool EnsureConfigInitializedCore() => true;
+#endif
+
+        internal static bool EnsureConfigInitialized() => EnsureConfigInitializedCore();
```

Comment:
```suggestion
#if !CORECLR
        private static bool EnsureConfigInitialized() => true;
#endif
```
Dtto for other instances of `EnsureConfigInitializedCore`